rust-runtime/aws-smithy-http/src/byte_stream.rs

```diff
@@ -609,17 +610,22 @@ mod tests {
         for i in 0..10000 {
             writeln!(file, "Brian was here. Briefly. {}", i)?;
         }
+        let file_length = file
+            .as_file()
+            .metadata()
+            .expect("file metadata is accessible")
+            .len();
+
         let body = ByteStream::read_from()
             .path(&file)
             .buffer_size(16384)
-            // This isn't the right file length - one shouldn't do this in real code
-            .file_size(200)
+            .length(file_length)
             .build()
             .await?
             .into_inner();
 
-        // assert that the file length specified size is used as size hint
-        assert_eq!(body.size_hint().exact(), Some(200));
```

Comment:
because we loaded the real file, this doesn't test that the hinting actually works anymore

Reply:
I wrote a new test just for the hinting and I think it's correct but I would appreciate your scrutiny